src/components/EmployeeList/EmployeeList.tsx

```diff
@@ -96,14 +100,67 @@ const EmployeeListItem: Component<EmployeeRow> = ({ employee, contracts, actions
 
 const EmployeeList: Component<Props> = ({ state, actions }) => {
   const { filterString } = state.view.employees;
-  const employees = state.employee.list || [];
+  const employees = state.employee.list;
   const contracts = state.contract.list;
+  const allocations = state.allocation.list;
+  const projects = state.project.list;
+
   const filteredEmployees = filterEmployees(employees, filterString);
 
+  const employeesProjectMap: Map<number, Set<ProjectModel>> = new Map();
+  const employeesContractMap: Map<number, ContractModel | undefined> = new Map();
+
+  allocations.forEach((allocation) => {
+    const { projectId, contractId } = allocation;
+
+    const contract = contracts.find(c => c.id === contractId);
+    const project = projects.find(p => p.id === projectId);
+
+    if (contract && project) {
+      const employeeId = contract.employeeId;
+      if (!employeesProjectMap.has(employeeId)) {
+        employeesProjectMap.set(employeeId, new Set())
+      }
+
+      const set = employeesProjectMap.get(employeeId)!;
+      set.add(project);
+    }
+  });
+
+  projects.forEach((project) => {
+    const { projectManagerId } = project;
+
+    if (project) {
+      if (!employeesProjectMap.has(projectManagerId)) {
+        employeesProjectMap.set(projectManagerId, new Set())
+      }
+
+      const set = employeesProjectMap.get(projectManagerId)!;
+      set.add(project);
+    }
+  });
+
+  employees.forEach((employee) => {
+    const sortedContracts = contracts.filter(c => c.employeeId == employee.id).sort((a, b) => moment(a.endDate, "YYY-MM-DD").diff(moment(b.endDate, "YYY-MM-DD")))
```

Comment:
1. use constants for the date format.
2. improve readability by indenting chaining:
3. move sort comparison to a util function

```
const compareByDate = (a: ContractModel, b: ContractModel): number {
  return moment(a.endDate, "YYY-MM-DD").diff(moment(b.endDate, "YYY-MM-DD")
}

contracts
  .filter(c => c.employeeId == employee.id)
  .sort((a, b) => compareByDate(a,b))
```

Reply:
even better, as dates are now Moments
```
const sortedContracts = contracts
                              .filter(c => c.employeeId == employee.id)
                              .sort((a, b) => a.endDate.diff(b.endDate))
```